src/app/components/transcriptions/view-transcript/view-transcript.component.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { HttpClientModule } from '@angular/common/http';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ActivatedRoute } from '@angular/router';
+import { CaseService } from '@services/case/case.service';
+import { TranscriptionService } from '@services/transcription/transcription.service';
+
+import { ViewTranscriptComponent } from './view-transcript.component';
+
+describe('ViewTranscriptComponent', () => {
+  let component: ViewTranscriptComponent;
+  let fixture: ComponentFixture<ViewTranscriptComponent>;
+
+  const mockActivatedRoute = {
+    snapshot: {
+      data: {
+        userState: {
+          userId: 123,
+          userName: 'dev@local',
+          roles: [
+            {
+              roleId: 123,
+              roleName: 'local dev',
+              permissions: [
+                {
+                  permissionId: 1,
+                  permissionName: 'local dev permissions',
+                },
+              ],
+            },
+          ],
+        },
+      },
+      params: {
+        caseId: '1',
+        hearing_id: '1',
+        transcriptId: '2',
+      },
+      queryParams: { tab: 'Transcripts' },
+    },
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [ViewTranscriptComponent, HttpClientModule],
+      providers: [
+        { provide: ActivatedRoute, useValue: mockActivatedRoute },
+        { provide: CaseService, useValue: { getCase: jest.fn() } },
+        { provide: TranscriptionService, useValue: { getTranscriptionDetails: jest.fn() } },
+      ],
+    });
+
+    fixture = TestBed.createComponent(ViewTranscriptComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should set the Case ID and Transcript ID', () => {
+    expect(component.caseId).toEqual('1');
+    expect(component.transcriptId).toEqual('2');
+  });
+
+  it('should call the correct getCase method with ID', () => {
+    const spy = jest.spyOn(component.caseService, 'getCase');
+    fixture.detectChanges();
+    expect(spy).toHaveBeenCalledWith('1');
+  });
+
+  it('should call the correct getTranscriptionDetails method with ID', () => {
+    const spy = jest.spyOn(component.transcriptionService, 'getTranscriptionDetails');
+    fixture.detectChanges();
+    expect(spy).toHaveBeenCalledWith('2');
+  });
```

Comment:
Could probably combine these 2 tests and the `should create` into one test?

Reply:
These two tests have been combined and put into the should create in an incoming commit